### ISSUE TITLE
Unsetting undefined variable is not an issue

### ISSUE THREAD

**Issue by jrfnl** [sirbrillig/phpcs-variable-analysis#174]
Trying to unset an undefined variable is not an issue in PHP and will not result in any warnings/notices: https://3v4l.org/XsK20

The following code snippet:
```php
function foo($array) {
	foreach ($array as $value) {
	}
	unset($key, $value);
}
```

... currently results in the following warning for using `$key` in `unset()`:
```
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 6 | WARNING | Variable $key is undefined.
----------------------------------------------------------------------
```

I'd like to propose to either not throw an error for undefined variables being used in `unset()`  or to give this a separate error code so those kind of warnings can be silenced.

**Comment by sirbrillig** [comment]
> Trying to unset an undefined variable is not an issue in PHP

That's good to know, but I wonder if it might still be a sign that there's a typo. For example:

```php
function foo($array) {
  unset($ary['value']);
  return $array;
}
```

In that case, I'd want to be warned that I'm using a variable that's undefined. Is there a real use-case where this would be problematic?

That said, I'd be happy to split this case into its own sniff code if it would help someone.

**Comment by jrfnl** [comment]
> In that case, I'd want to be warned that I'm using a variable that's undefined.

Good point.

> I'd be happy to split this case into its own sniff code if it would help someone.

:+1: 

> Is there a real use-case where this would be problematic?

Yes, I can think of a case where a variable would be conditionally unset, with an `isset()` later on to determine whether to continue or not. A typo in that case would break the logic.

Fake, nonsense code to demonstrate:
```php
function foo() {
    $a = null;
    $b = null;
    $array = function_call();
    list($a, $b) = $array;
    if ($a < 20) {
        unset($aa); // Typo
    }
    if ($b < 20) {
        unset($b);
    }

    if (isset($a, $b) === false) {
        return;
    }

    // Do something.
```
